### PR TITLE
добавлена задача 7_1

### PR DESCRIPTION
**Pull request by evgkrylova**

### CODE REVIEW

**Comment by jzuken** [comment]
Your command: 1
Enter the value you want to add: 1
The process was successfully completed.

Your next command: 1
Enter the value you want to add: 5
The process was successfully completed.

Your next command: 1
Enter the value you want to add: 4
The process was successfully completed.

Your next command: 1
Enter the value you want to add: 6
The process was successfully completed.

Your next command: 1
Enter the value you want to add: 0
The process was successfully completed.

Your next command: 6
(1 (0 null null) (5 (4 null null) (6 null null)))
The process was successfully completed.

Your next command: 2
Enter the value you want to remove: 1


и всё упало

File: sem1/hw7/7_1/7_1.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#include <iostream>
+#include "binarySearchTree.h"
+
+using namespace std;
+
+int main()
+{
+	cout << "Hello. This programme works with data structures called 'Binery Search Tree'.\n";
```
Comment (jzuken) [inline sem1/hw7/7_1/7_1.cpp]: binary

File: sem1/hw7/7_1/7_1.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	cout << "\t 3 - to know if a value can be found in the tree;\n";
+	cout << "\t 4 - to print all existing values in ascending order;\n";
+	cout << "\t 5 - to print all existing values in descending order;\n";
+	cout << "\t 6 - to print all existing values the way the tree is stored.\n";
+	cout << "\nYour command: ";
+
+	int choice = -1;
+	cin >> choice;
+
+	BinaryTree *tree = createBinaryTree();
```
Comment (jzuken) [inline sem1/hw7/7_1/7_1.cpp]: вообще по условию множество просили. которое реализуется через дерево, да, но пользователь должен работать со множеством, а не с деревом

File: sem1/hw7/7_1/7_1.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+	while (choice != 0)
+	{
+		int value = 0;
+
+		if (choice == 1)
```
Comment (jzuken) [inline sem1/hw7/7_1/7_1.cpp]: лучше использовать switch тут

File: sem1/hw7/7_1/binarySearchTree.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+BinaryTree *createBinaryTree();
+BinaryTreeNode *createBinaryTreeNode(int value);
+
+void deleteNode(BinaryTreeNode *&node);
+void deleteTree(BinaryTree *&tree);
+
+void addValueToNode(BinaryTreeNode *&node, int value);
+void addValueToTree(BinaryTree *tree, int value);
+
+void deleteValueFromNode(BinaryTreeNode *&node, int value);
```
Comment (jzuken) [inline sem1/hw7/7_1/binarySearchTree.h]: а эти функции для работы с нодами точно нужны в публичном интерфейсе модуля?

**Comment by jzuken** [comment]
Your next command: 6
(1 null (4 (2 null (3 null null)) (7 (6 null null) (8 null null))))
The process was successfully completed.

Your next command: 2
Enter the value you want to remove: 1
The process was successfully completed.

Your next command: 6
(4 (2 null (3 null null)) (7 (6 null null) (8 null null)))
The process was successfully completed.

Your next command: 2
Enter the value you want to remove: 4
The process was successfully completed.

Your next command: 6
(3 (2 null null) (7 (6 null null) (8 null null)))
The process was successfully completed.


тут же нет балансировки, почему оно так недетерминированно удаляет?

**Comment by jzuken** [comment]
и там при удалении память теряется

**Comment by evgkrylova** [comment]
@jzuken при удаление вершины она ставит на её место большую из меньших. мы же вроде и на паре так же разбирали

**Pull request merged**